pmdarima/arima/tests/test_arima.py

```diff
@@ -1065,3 +1066,64 @@ def test_issue_191():
         alpha=0.05,
         suppress_warnings=True,
         trace=True)
+
+
+# test StepwiseContext parameter validation
+@pytest.mark.parametrize(
+    'max_steps,max_dur', [
+        pytest.param(-1, None),
+        pytest.param(0, None),
+        pytest.param(1001, None),
+        pytest.param(1100, None),
+        pytest.param(None, -1),
+        pytest.param(None, 0),
+    ])
+def test_stepwise_context_args(max_steps, max_dur):
+    with pytest.raises(ValueError):
+        StepwiseContext(max_steps=max_steps, max_dur=max_dur)
+
+
+# test auto_arima stepwise run with  StepwiseContext
+def test_auto_arima_with_stepwise_context():
+    samp = lynx[:8]
+    with StepwiseContext(max_steps=5, max_dur=30):
+        with pytest.warns(UserWarning) as uw:
+            auto_arima(samp, suppress_warnings=False, stepwise=True,
+                       error_action='ignore')
+
+            # assert that max_steps were taken
+            assert any(str(w.message)
+                       .startswith('stepwise search has reached the '
+                                   'maximum number of tries') for w in uw)
+
+
+# test effective context info in nested context scenario
+def test_nested_context():
+    ctx1_data = {'max_dur': 30}
+    ctx2_data = {'max_steps': 5}
+    ctx1 = StepwiseContext(**ctx1_data)
+    ctx2 = StepwiseContext(**ctx2_data)
+
+    with(ctx1):
+        with(ctx2):
+            effective_ctx_data = ContextStore.get_or_empty(ContextType.STEPWISE)
+            expected_ctx_data = ctx1_data.copy()
+            expected_ctx_data.update(ctx2_data)
+
+            assert all(effective_ctx_data[key] == expected_ctx_data[key]
+                       for key in expected_ctx_data.keys())
+
+            assert all(effective_ctx_data[key] == expected_ctx_data[key]
+                       for key in effective_ctx_data.keys())
```

Comment:
Is this asserting the same thing as above?

Reply:
I'm trying to assert that: _effective &bigcap; expected_ == _expected &bigcap; effective_